### PR TITLE
samd/main.c: Add /lib to sys.path.

### PR DESCRIPTION
**Pull request by jimmo**
Needed for mip to find a default location to install to.

_This work was funded through GitHub Sponsors._

### CODE REVIEW

**Comment by robert-hh** [comment]
The flash file system of the MIMXRT port is mounted at `/flash`. So you have to add `/flash/lib`.

**Comment by jimmo** [comment]
Ah thanks, I missed that mimxrt handles this in `_boot.py`.

I've made this PR just do SAMD. I agree, it would be simple to do this in _boot.py also, but because ESP32 does it in main.c I will keep SAMD the same as ESP32.

At some stage I would really like to make `/` vs `/flash` consistent on all ports. Maybe this can be a MicroPython 2.0 thing, because we need to do a re-think of flash+partitions anyway to support the MapFS feature (#8381)

**Comment by jimmo** [comment]
Updated to use _boot.py. We should fix ESP32 at some stage (e.g. as part of a bigger consolidation of flash/mounting/storage across ports).

**Comment by robert-hh** [comment]
It is common for "older" ports like ESP32 and STM32 that they make settings to the Python environment in main.c which could be cleaner and shorter made in _boot,py. That's just the history.

File: ports/samd/modules/_boot.py [hunk 1]
```diff
@@ -13,7 +14,8 @@
     fs_type.mkfs(bdev, progsize=256)
     vfs = fs_type(bdev, progsize=256)
 os.mount(vfs, "/")
+sys.path.append("/lib")
 
-del vfs, fs_type, bdev, os, samd
+del vfs, fs_type, bdev, os, samd, sys
```
Comment (dpgeorge) [inline ports/samd/modules/_boot.py:19]: I added sys to the deleted globals.

**Pull request merged**